src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,75 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useContext } from 'react';
+import cn from 'classnames';
+import { TodoList } from '../TodoList/TodoList';
+import { TodosFilter } from '../TodosFilter/TodosFilter';
+import { TodoContext } from '../TodoContext/TodoContext';
+
+export const TodoApp: React.FC = () => {
+  const { todos, error, setError } = useContext(TodoContext);
+
+  const anyCompletedTodos = todos.some(todo => todo.completed);
+  const anyActiveTodos = todos.some(todo => !todo.completed);
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this buttons is active only if there are some active todos */}
+          {anyActiveTodos && (
+            <button type="button" className="todoapp__toggle-all active" />
+          )}
+
+          {/* Add a todo on form submit */}
+          <form>
+            <input
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+            />
+          </form>
+        </header>
+
+        {todos.length > 0 && (
+          <>
+            <TodoList />
+
+            {/* Hide the footer if there are no todos */}
+            <footer className="todoapp__footer">
+              <span className="todo-count">
+                3 items left
+              </span>
+
+              <TodosFilter />
+
+              {/* don't show this button if there are no completed todos */}
+              {anyCompletedTodos && (
+                <button type="button" className="todoapp__clear-completed">
+                  Clear completed
+                </button>
```

Comment:
on the work version, this button shows all time, and if you don't show this element then other filter buttons move to the right side of the footer